examples/kubeflow/mnist/pipeline.py

```diff
@@ -82,21 +88,124 @@ def kubeflow_pipeline(s3_endpoint: str, input_bucket: str):
+                    'pipeline. Used to create a pipeline from a local machine.')
+parser.add_argument('--force', action='store_true', default=False,
+                    help='If set, and this script tries to create a pipeline '+
+                    'where a kubeflow pipeline with the same name already '+
+                    'exists, the existing pipeline will be deleted')
+
+def pipeline_id(client : kfp.Client, name : str):
+    """Gets the ID of the kubeflow pipeline with the name 'name'
+    Args:
+      name of the pipeline
+    Returns:
+      id of the pipeline
+    """
+    page_token = ""
+    while True:
+        p = client.list_pipelines(page_token=page_token, page_size=100)
+        if p.pipelines is None:
+            return ""
+        for p in p.pipelines:
+            if p.name == name:
+                return p.id
+        if p.next_page_token is None:
+            return ""
+        page_token = p.next_page_token
+
+def experiment_id(client : kfp.Client, name : str):
+    """Gets the ID of the kubeflow experiment with the name 'name'
+    Args:
+      name of the experiment
+    Returns:
+      id of the experiment
+    """
+    page_token = ""
+    while True:
+        p = client.list_experiments(page_token=page_token, page_size=100)
+        for p in p.experiments:
+            if p.name == name:
+                return p.id
+        if p.next_page_token is None:
+            return ""
+        page_token = p.next_page_token
 
-    print("run id: {}".format(run_id))
-
-    j = client.wait_for_run_completion(run_id, 60)
-    assert j.run.status == 'Succeeded'
+def main():
+    args = parser.parse_args()
+    if args.remote_host != "":
+        client = kfp.Client(host=args.remote_host)
+    else:
+        client = kfp.Client()
+
+    run_id = ""
+    if args.create_run_in != "" and args.create_pipeline != "":
+        logging.error('only one of --create_run_in and '+
+              '--create_pipeline may be set')
+        sys.exit(1)
+    if args.create_run_in != "":
+        logging.info('creating run in pipeline "{}"'.format(args.create_run_in))
+        pid = pipeline_id(client, args.create_run_in)
+        if pid == "":
+            logging.error('could not find pipeline "{}" to create job'.format(
+                args.create_run_in))
+            sys.exit(1)
+        # Create a run in the target pipeline using the new pipeline ID
+        run_info = client.run_pipeline(
+            job_name="pach-job-" + os.environ["PACH_JOB_ID"],
+            pipeline_id=pid,
+            experiment_id=experiment_id(client, "Default"),
+            params = {
+                "s3_endpoint": os.environ["S3_ENDPOINT"],
+                "input_bucket": "input",
+            }
+        )
+        run_id = run_info.id
+    elif args.create_pipeline != "":
+        # Local machine is just creating the pipeline
+        try:
+            (_, pipeline_package_path) = tempfile.mkstemp(suffix='.zip')
```

Comment:
Use `tempfile.NamedTemporaryFile` as a context manager to guarantee cleanup, e.g. see here https://docs.python.org/3/library/tempfile.html#examples

Reply:
:+1: I also had to pass the `.name` field to the KF client, which I *think* was correct:
```
with tempfile.NamedTemporaryFile(suffix='.zip') as pipeline_file:
    compiler.Compiler().compile(kubeflow_pipeline, pipeline_file.name)
```